src/main/java/Duke.java

```diff
@@ -1,10 +1,303 @@
+
+    public static String goodbye = "____________________________________________________________\n"
+            + "Bye. Hope to see you again soon!\n"
+            + "____________________________________________________________\n";
+
+
+    public static String emptyTaskError = "____________________________________________________________\n"
+            + "Please do not leave the description of the task empty!\n"
+            + helpMessage
+            + "____________________________________________________________\n";
+
+    public static String invalidTaskError = "____________________________________________________________\n"
+            + "I don't know what that means! It's either an invalid command or you have formatted the task wrongly.\n"
+            + helpMessage
+            + "____________________________________________________________\n";
+
+
+    public static ArrayList<Task> taskList = new ArrayList<>();
+
+    public static String fileNotFound = "____________________________________________________________\n"
+            + "No preloaded file found! Please input your own data.\n"
+            + "____________________________________________________________\n";
+
+    private static char getTaskType(Task task) {
+        if (task instanceof Todo) {
+            return 'T';
+        }
+        if (task instanceof Deadline) {
+            return 'D';
+        }
+        return 'E';
+    }
+
+    private static void updateData(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath); //to write first line
+        FileWriter fa = new FileWriter(filePath, true); //append the rest
+
+        if (taskList.isEmpty()) {
+            fw.close();
+            fa.close();
+            return;
+        }
+
+        Task currentTask = taskList.get(0);
+        char taskType = getTaskType(currentTask);
+        int isDone = (currentTask.getStatus()) ? 1 : 0;
+        String fullTask = currentTask.toString().substring(7);
+
+        fw.write(taskType + " | " + isDone + " | " + fullTask + System.lineSeparator());
+
+        for (int i = 1; i < taskList.size(); i++) {
+            currentTask = taskList.get(i);
+            taskType = getTaskType(currentTask);
+            isDone = (currentTask.getStatus()) ? 1 : 0;
+            fullTask = currentTask.toString().substring(7);
+            fa.write(taskType + " | " + isDone + " | " + fullTask + System.lineSeparator());
+        }
+
+        fw.close();
+        fa.close();
+    }
+
+    private static void readData(String filePath) throws FileNotFoundException {
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+
+            String taskInfo = s.nextLine();
+
+            if (taskInfo.startsWith("T")) {
+                Task newTask = new Todo(taskInfo.substring(8));
+                taskList.add(newTask);
+                if (taskInfo.charAt(4) == '1') {
```

Comment:
perhaps remove theses additional magic number here? likely to check these cases in other places as well :)